WcaOnRails/app/models/user.rb

```diff
@@ -298,6 +298,13 @@ def senior_delegate_presence
     end
   end
 
+  validate :must_have_senior_delegate
+  def must_have_senior_delegate
+    if User.delegate_status_allows_senior_delegate(delegate_status) && !senior_delegate
+      errors.add(:senior_delegate, I18n.t('users.errors.must_be_present'))
+    end
+  end
+
```

Comment:
I believe you can use Rails build-in validation, which wouldn't require new translation phrase.
Something along the lines of:
```rb
validates :senior_delegate, presence: true, if: -> { User.delegate_status_allows_senior_delegate(delegate_status) && !senior_delegate }
```
This should result in the same error message, but already translated =)

Reply:
That's really neat!

Reply:
@AlbertoPdRF, with the work you're doing in this PR, should we rename `delegate_status_allows_senior_delegate` to `delegate_status_requires_senior_delegate`?

Reply:
Sounds like a good change to me, should I proceed?

Reply:
Go for it!